v4/export/prepare.go

```diff
@@ -0,0 +1,160 @@
+package export
+
+import (
+	"database/sql"
+	"errors"
+	"strings"
+)
+
+func detectServerInfo(db *sql.DB) (ServerInfo, error) {
+	versionStr, err := SelectVersion(db)
+	if err != nil {
+		return ServerInfoUnknown, err
+	}
+	return ParseServerInfo(versionStr), nil
+}
+
+func prepareDumpingDatabases(conf *Config, db *sql.DB) ([]string, error) {
+	if conf.Database == "" {
+		return ShowDatabases(db)
+	} else {
+		return strings.Split(conf.Database, ","), nil
+	}
+}
+
+func listAllTables(db *sql.DB, databaseNames []string) (DatabaseTables, error) {
+	dbTables := DatabaseTables{}
+	for _, dbName := range databaseNames {
+		err := UseDatabase(db, dbName)
+		if err != nil {
+			return nil, err
+		}
+		tables, err := ShowTables(db)
+		if err != nil {
+			return nil, err
+		}
+		dbTables[dbName] = tables
+	}
+	return dbTables, nil
+}
+
+type Consistency int8
```

Comment:
IMO `Consistency` looks more like an `interface` than an enum (in particular you're calling it a "strategy") 🤷.

```go
type Consistency interface {
    setupConsistency(*ConsistencyController) error
    tearDownConsistency(*ConsistencyController) error
}
```